src/sentry/issues/endpoints/browser_reporting_collector.py

```diff
@@ -17,30 +18,46 @@
 
 logger = logging.getLogger(__name__)
 
-# Known browser report types as defined by the Browser Reporting API specification
-BrowserReportType = Literal[
-    # Core report types (always sent to 'default' endpoint)
-    "deprecation",  # Deprecated API usage
-    "intervention",  # Browser interventions/blocks
-    "crash",  # Browser crashes
-    # Policy violation report types (can be sent to named endpoints)
-    "csp-violation",  # Content Security Policy violations
-    "coep",  # Cross-Origin-Embedder-Policy violations
-    "coop",  # Cross-Origin-Opener-Policy violations
-    "document-policy-violation",  # Document Policy violations
-    "permissions-policy",  # Permissions Policy violations
+BROWSER_REPORT_TYPES = [
+    "deprecation",
+    "intervention",
+    "crash",
+    "csp-violation",
+    "coep",
+    "coop",
+    "document-policy-violation",
+    "permissions-policy",
 ]
 
 
-@dataclass
-class BrowserReport:
-    body: dict[str, Any]
-    type: BrowserReportType
-    url: str
-    user_agent: str
-    destination: str
-    timestamp: int
-    attempts: int
+# Working Draft https://www.w3.org/TR/reporting-1/#concept-reports
+# Editor's Draft https://w3c.github.io/reporting/#concept-reports
+# We need to support both
+class BrowserReportSerializer(serializers.Serializer[Any]):
+    """Serializer for validating browser report data structure."""
+
+    body = serializers.DictField()
+    type = serializers.ChoiceField(choices=BROWSER_REPORT_TYPES)
+    url = serializers.URLField()
+    user_agent = serializers.CharField()
+    destination = serializers.CharField()
+    attempts = serializers.IntegerField(min_value=1)
+    # Fields that do not overlap between specs
+    # We need to support both specs
+    age = serializers.IntegerField(required=False)
+    timestamp = serializers.IntegerField(required=False, min_value=0)
+
+    def validate_timestamp(self, value: int) -> int:
```

Comment:
These two methods validate that we're dealing with the format for one spec or the other.